src/semaphore.rs

```diff
@@ -72,16 +86,15 @@ impl Into<crate::gl::types::GLenum> for TextureLayout {
 /// Similar to a GL sync object, this describes a semaphore which can be used for OpenGL-Vulkan command queue synchronization.
 pub struct Semaphore {
     context: Rc<Context>,
-    id: gl::types::GLuint,
-    backing_fd: Option<File>,
+    id: gl::types::GLuint,    
 }
 
 impl Semaphore {
     /// Creates a semaphore imported from an opaque file descriptor.
     #[cfg(target_os = "linux")]
     pub unsafe fn new_from_fd<F: Facade + ?Sized>(
         facade: &F,
-        fd: File,
+        fd: std::fs::File,
```

Comment:
The problem is that since this function is specific to Linux (Windows requires another extension to be used), if I import this for the entire module the Windows and MacOS builds will have warnings. In order to avoid that I either have to import this only for the function or import it for the entire module and add an annotation to ignore the warning. Which solution would be preferable?

Reply:
That's a good point. Resolved.